tools/python_api/src_cpp/py_connection.cpp

```diff
@@ -56,13 +67,105 @@ py::str PyConnection::getNodePropertyNames(const std::string& tableName) {
     return conn->getNodePropertyNames(tableName);
 }
 
+py::str PyConnection::getNodeTableNames() {
+    return conn->getNodeTableNames();
+}
+
+py::str PyConnection::getRelPropertyNames(const std::string& tableName) {
+    return conn->getRelPropertyNames(tableName);
+}
+
+py::str PyConnection::getRelTableNames() {
+    return conn->getRelTableNames();
+}
+
 PyPreparedStatement PyConnection::prepare(const std::string& query) {
     auto preparedStatement = conn->prepare(query);
     PyPreparedStatement pyPreparedStatement;
     pyPreparedStatement.preparedStatement = std::move(preparedStatement);
     return pyPreparedStatement;
 }
 
+uint64_t PyConnection::getNumNodes(const std::string& nodeName) {
+    return storageDriver->getNumNodes(nodeName);
+}
+
+uint64_t PyConnection::getNumRels(const std::string& relName) {
+    return storageDriver->getNumRels(relName);
+}
+
+void PyConnection::getAllEdgesForTorchGeometric(py::array_t<int64_t>& npArray,
+    const std::string& srcTableName, const std::string& relName, const std::string& dstTableName,
+    size_t queryBatchSize) {
+    // Get the number of nodes in the src table for batching.
+    auto numSrcNodes = getNumNodes(srcTableName);
+    int64_t batches = numSrcNodes / queryBatchSize;
+    if (numSrcNodes % queryBatchSize != 0) {
+        batches += 1;
+    }
+    auto numRels = getNumRels(relName);
+
+    auto buffer_info = npArray.request();
+    auto buffer = (int64_t*)buffer_info.ptr;
+    std::cout << "Done getting buffer for result numpy arrays." << std::endl;
+
+    // Run queries in batch to fetch edges.
+    auto queryString = "MATCH (a:{})-[:{}]->(b:{}) WHERE offset(id(a)) >= $s AND offset(id(a)) < "
+                       "$e RETURN offset(id(a)), offset(id(b))";
+    auto query = StringUtils::string_format(queryString, srcTableName, relName, dstTableName);
+    auto preparedStatement = conn->prepare(query);
+    for (int64_t batch = 0; batch < batches; ++batch) {
+        int64_t start = batch * queryBatchSize;
+        int64_t end = (batch + 1) * queryBatchSize;
+        end = end > numSrcNodes ? numSrcNodes : end;
+        std::unordered_map<std::string, std::shared_ptr<Value>> parameters;
+        parameters["s"] = std::make_shared<Value>(start);
+        parameters["e"] = std::make_shared<Value>(end);
+        std::cout << "Batch " << batch << " from " << start << " to " << end << std::endl;
```

Comment:
Maybe we should remove before the final release. But let's keep it for benchmark purpose